src/java.base/share/classes/sun/security/ssl/ECDHClientKeyExchange.java

```diff
@@ -269,22 +269,9 @@ public void consume(ConnectionContext context,
                     "No expected EC server cert for ECDH client key exchange");
             }
 
-            // Determine which NamedGroup we'll be using, then use
-            // the creator functions.
-            NamedGroup namedGroup = null;
-
             // Iteratively determine the X509Possession type's ParameterSpec.
             ECParameterSpec ecParams = x509Possession.getECParameterSpec();
-            NamedParameterSpec namedParams = null;
-            if (ecParams != null) {
-                namedGroup = NamedGroup.valueOf(ecParams);
-            }
-
-            // Wasn't EC, try XEC.
-            if (ecParams == null) {
-                namedParams = x509Possession.getXECParameterSpec();
-                namedGroup = NamedGroup.nameOf(namedParams.getName());
-            }
+            NamedParameterSpec namedParams = x509Possession.getXECParameterSpec();
```

Comment:
It may not necessary to define 'ecParams' and 'namedParams' any longer, which was used to find out the named group.  Now, the checking could be placed on the "namedGroup" (if the named group is EC/CDH) around line 293.